src/Nethermind/Nethermind.Blockchain/Blocks/BlockStore.cs

```diff
@@ -36,6 +40,18 @@ public void SetMetadata(byte[] key, byte[] value)
         return _blockDb.Get(key);
     }
 
+    private void TruncateToMaxSize()
+    {
+        int toDelete = (int)(_blockDb.GetSize() - _maxSize!);
+        if (toDelete > 0)
+        {
+            foreach (var blockToDelete in GetAll().OrderBy(block => block.Number).Take(toDelete))
```

Comment:
The downside here is that you need to load all the blocks to delete them, would be better if we could store the blocks in this order, and then we could just delete a few from the beginning. Not sure if worth investigating it though.

Reply:
Ah it looks like we prepend keys with block number so we can do that!

Reply:
Yeah seems like the blocks should already be sorted correctly